### ISSUE TITLE
Extract statement in slow query module not working

### ISSUE THREAD

**Issue by nbozic-zz** [smartcat-labs/cassandra-diagnostics#93]

**Comment by nbozic-zz** [comment]
When slow query is logged only statement type is extracted and in influx we see empty string for statement.

**Comment by f1yegor** [comment]
I've looked at issue and with current pointcut was able to reconstruct SelectStatement. But I'm thinking about another, more convenient pointcut for this purpose. Your ideas?

**Comment by mgobec** [comment]
@f1yegor Could you please elaborate? I'm not sure I understand. The issue here is that when we create query object in connector we set [empty string](https://github.com/smartcat-labs/cassandra-diagnostics/blob/dev/cassandra-diagnostics-connector21/src/main/java/io/smartcat/cassandra/diagnostics/connector/QueryProcessorWrapper.java#L105) to statement instead of the statement itself. We didn't have time to add this but any help is appreciated.

**Comment by f1yegor** [comment]
Something like
`

```
private final CQLStatement statement;

QueryBuilder(CQLStatement statement) {
    this.statement = statement;
}

String build() {
    if (statement instanceof SelectStatement) {
        SelectStatement select = (SelectStatement) statement;
        Selection selection = select.getSelection();
        String columnsQuery = "";
        if (selection != null) {
            List<ColumnDefinition> columns = select.getSelection().getColumns();
            List<String> columnNames = new ArrayList<>(columns.size());
            for (ColumnDefinition column : columns) {
                columnNames.add(column.name.toCQLString());
            }
            columnsQuery = StringUtils.join(columnNames, ", ");
        }
        return "SELECT " + columnsQuery + " FROM " + select.keyspace() + "." + select.columnFamily();
    }
    return "";
}
```

`
and then use as statement query.

**Comment by mgobec** [comment]
This looks like an extra effort just to get the statement which we already have at process method in QueryProcessor and there would need to be a significant effort to implement full select or upsert statements because of all the parameters. For example your code is missing WHERE params which are useful when investigating slow queries.
I think @nivancevic did an investigation and we should be able to use the queryString parameter from process method and provide it to the wrapper using bytebuddy. @nivancevic Any comments on this?

**Comment by f1yegor** [comment]
Yes, I missed WHERE for further implementation.
Looked at implementation of ByteBuddy. It would be challenging or impossible, especially because `processStatement` called from several places. Would be interesting to look at good solution.

**Comment by nbozic-zz** [comment]
Fixed in #173 where we moved whole logic from ByteBuddy byte code manipulation to custom QueryHandler.

@f1yegor Check out release candidate [2.0.0-rc1](https://bintray.com/smartcat-labs/maven/cassandra-diagnostics/2.0.0-rc1) on BinTray.
